company-ycmd.el

```diff
@@ -519,6 +526,8 @@ If CB is non-nil, call it with candidates."
                    (get-text-property 0 'params candidate))
     (when (memq major-mode '(python-mode rust-mode))
       (setq it (company-ycmd--remove-self-from-function-args it)))
+    (when (eq major-mode 'rust-mode)
```

Comment:
You can move the second `when` into the first `when` form because it already checks for `rust-mode`. In non rust modes, we would still have only one check.

```emacs-lisp
     (when (memq major-mode '(python-mode rust-mode))
        (setq it (company-ycmd--remove-self-from-function-args it))
        (when (eq major-mode 'rust-mode)
           (setq it (company-ycmd--remove-template-args-from-function-args it)))
)
```